server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java

```diff
@@ -342,20 +343,19 @@ public static void importRealm(KeycloakSession session, RealmRepresentation rep,
         if (rep.getClients() != null) {
             for (ClientRepresentation resourceRep : rep.getClients()) {
                 if (resourceRep.getDefaultRoles() != null) {
-                    ClientModel clientModel = newRealm.getClientByClientId(resourceRep.getClientId());
+                    ClientModel clientModel = createdClients.getOrDefault(resourceRep.getClientId(), newRealm.getClientByClientId(resourceRep.getClientId()));
```

Comment:
Isn't better `computIfAbsent`? I mean, the way you did you still have the call to `newRealm.getClientById`.

Reply:
Thanks for the review. You are right. changed it to computeIfAbsent